pex/network_configuration.py

```diff
@@ -0,0 +1,58 @@
+# coding=utf-8
+# Copyright 2020 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import
+
+from collections import namedtuple
+
+
+class NetworkConfiguration(namedtuple('NetworkConfiguration', [
+  'cache_ttl',
+  'retries',
+  'timeout',
+  'headers',
+  'proxy',
+  'cert',
+  'client_cert'
+])):
+  """Configurtion for network requests made by the resolver."""
+
+  @classmethod
+  def create(
+      cls,
+      cache_ttl=3600,
+      retries=5,
+      timeout=15,
+      headers=None,
+      proxy=None,
+      cert=None,
+      client_cert=None
+  ):
+    """
+    Creates a new network configuration accepting defaults for any values not explicitly provided.
+
+    :param int cache_ttl: The maximum time an item is retained in the HTTP cache in seconds.
+    :param int retries: The maximum number of retries each connection should attempt.
+    :param timeout: The socket timeout in seconds.
+    :param headers: A list of additional HTTP headers to send in the form NAME:VALUE.
+    :type headers: list of str
+    :param str proxy: A proxy to use in the form [user:passwd@]proxy.server:port.
+    :param str cert: The path to an alternate CA bundle.
+    :param str client_cert: The path to an SSL client certificate which should be a single file
+                            containing the private key and the certificate in PEM format.
+    :return: The resulting network configuration.
+    :rtype: :class:`NetworkConfiguration`
+    """
+    assert cache_ttl >= 0, 'The cache_ttl parameter should be >= 0; given: {}'.format(cache_ttl)
+    assert retries >= 0, 'The retries parameter should be >= 0; given: {}'.format(retries)
+    assert timeout >= 0, 'The timeout parameter should be > 0; given: {}'.format(timeout)
```

Comment:
Safer to use `raise AssertionError`, as `assert` is not run when byte code is optimized (https://bandit.readthedocs.io/en/latest/plugins/b101_assert_used.html).

Reply:
I'm not too worried about that. I would like to beef up the API types though to do argument validation more thoroughly though at which point these will switch from asserts to at least ValueErrors. This is tracked in #969 with a note just added.